jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/reporters/JunitTestsReporter.java

```diff
@@ -159,27 +150,43 @@ public void process(@Nonnull StepContext context, @Nonnull Element mavenSpyLogsE
             return;
         }
 
+        List<Element> sureFireTestEvents = XmlUtils.getExecutionEvents(mavenSpyLogsElt, "org.apache.maven.plugins", "maven-surefire-plugin", "test");
+        List<Element> failSafeTestEvents = XmlUtils.getExecutionEvents(mavenSpyLogsElt, "org.apache.maven.plugins", "maven-failsafe-plugin", "integration-test");
+
+        executeReporter(context, listener, sureFireTestEvents);
+        executeReporter(context, listener, failSafeTestEvents);
+    }
+
+    private void executeReporter(StepContext context, TaskListener listener, List<Element> testEvents) throws IOException, InterruptedException {
+        FilePath workspace = context.get(FilePath.class); // TODO check that it's the good workspace
+        Run run = context.get(Run.class);
+        Launcher launcher = context.get(Launcher.class);
 
-        for (Element sureFireTestEvent : sureFireTestEvents) {
-            String surefireEventType = sureFireTestEvent.getAttribute("type");
+        if (testEvents.isEmpty()) {
+            LOGGER.log(Level.FINE, "No org.apache.maven.plugins:maven-surefire-plugin:test execution found");
```

Comment:
The reference to `maven-surefire-plugin:test` is wrong when we invoke `executeReporter` for `maven-failsafe-plugin:integration-test`.
Maybe `isEmpty()` should be invoked before each  `executeReporter(context, listener, sureFireTestEvents)`. It's a bit verbose but it would provide accurate detailed log messages.

As you have probably seen, I try to have very detailed log message as there may be many maven plugins and builds with behaviours that we have not anticipated.